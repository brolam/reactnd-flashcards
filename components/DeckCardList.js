import React from 'react'
import { StyleSheet } from 'react-native'
import { View, FlatList } from 'react-native'
import PropTypes from 'prop-types'
import DeckCard from '../components/DeckCard'

export default function DeckCardList({ decks }) {
  return (
    <View style={styles.container}>
      <FlatList
        data={decks}
        renderItem={({item}) => <DeckCard key={item.id} {...item} />}
      />
    </View>
  )
}

DeckCardList.propTypes = {
  decks: PropTypes.array.isRequired,
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: 22
  }
})