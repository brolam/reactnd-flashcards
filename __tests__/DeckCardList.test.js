import React from 'react'
import DeckCardList from '../components/DeckCardList'

test('renders without crashing', () => {
  const wrapper = shallow(<DeckCardList decks={deckDummies} />);
  expect(wrapper).toMatchSnapshot();
});

const deckDummies = [
  { id: 'one-item', title: 'One Deck', amountOfCards: 10 },
  { id: 'two-item', title: 'Two Deck', amountOfCards: 11 },
  { id: 'three-item',title: 'Three Deck', amountOfCards: 12 }
]