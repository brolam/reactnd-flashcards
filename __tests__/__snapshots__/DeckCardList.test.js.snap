// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingTop": 22,
    }
  }
>
  <FlatList
    data={
      Array [
        Object {
          "amountOfCards": 10,
          "title": "One Deck",
        },
        Object {
          "amountOfCards": 11,
          "title": "Two Deck",
        },
        Object {
          "amountOfCards": 12,
          "title": "Three Deck",
        },
      ]
    }
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</View>
`;
